lib/utils.js

```diff
@@ -766,3 +772,32 @@ function handleCustomlevelNamesOpts (cLevels) {
     return {}
   }
 }
+
+/**
+ * Translates all conditional blocks from within the messageFormat. Translates any matching
+ * {if key}{key}{end} statements and returns everything between if and else blocks if the key provided
+ * was found in log.
+ *
+ * @param {string} messageFormat A format string or function that defines how the
+ *  logged message should be conditionally formatted, e.g. `'{if level}{level}{end} - {if req.id}{req.id}{end}'`.
+ * @param {object} log The log object to be modified.
+ *
+ * @returns {string} The parsed messageFormat.
+ */
+function parseMessageFormat (messageFormat, log) {
+  messageFormat = messageFormat.replace(/{if (.*?)}(.*?){end}/g, function (_, key, value) {
+    const propertyValue = getPropertyValue(log, key)
+    if (propertyValue && value.includes(key)) {
+      return value.replace(new RegExp('{' + key + '}', 'g'), propertyValue)
+    } else {
+      return ''
+    }
+  })
+
+  // Remove unescaped if blocks
+  messageFormat = messageFormat.replace(/{if (.*?)}/g, '')
+  // Remove unescaped end blocks
+  messageFormat = messageFormat.replace(/{end}/g, '')
+
+  return messageFormat.replace(/\s+/g, ' ').trim()
+}
```

Comment:
I think this would be a bit easier to read:

```suggestion
function parseMessageFormat (messageFormat, log) {
  messageFormat = messageFormat.replace(/{if (.*?)}(.*?){end}/g, replacer)

  // Remove unescaped if blocks
  messageFormat = messageFormat.replace(/{if (.*?)}/g, '')
  // Remove unescaped end blocks
  messageFormat = messageFormat.replace(/{end}/g, '')

  return messageFormat.replace(/\s+/g, ' ').trim()
  
  function replacer (_, key, value) {
    const propertyValue = getPropertyValue(log, key)
    if (propertyValue && value.includes(key)) {
      return value.replace(new RegExp('{' + key + '}', 'g'), propertyValue)
    } else {
      return ''
    }
  }
}
```